network/network_test.go

```diff
@@ -306,6 +311,50 @@ func TestNewNetwork(t *testing.T) {
 	wg.Wait()
 }
 
+func TestIngressConnCount(t *testing.T) {
+	require := require.New(t)
+
+	emptyHandler := func(context.Context, message.InboundMessage) {}
+
+	_, networks, wg := newFullyConnectedTestNetwork(
+		t, []router.InboundHandler{
+			router.InboundHandlerFunc(emptyHandler),
+			router.InboundHandlerFunc(emptyHandler),
+			router.InboundHandlerFunc(emptyHandler),
+		})
+
+	wg.Done()
+
+	for _, net := range networks {
+		net.config.NoIngressValidatorConnectionGracePeriod = 0
+		net.config.HealthConfig.Enabled = true
+	}
+
+	require.Eventually(func() bool {
+		result := true
+		for _, net := range networks {
+			result = result && len(net.PeerInfo(nil)) == len(networks)-1
+		}
+		return result
+	}, time.Minute, time.Millisecond*10)
+
+	ingressConnections := make([]int, 0, len(networks))
+	healthCheckErrors := make([]error, 0, len(networks))
+
+	for _, net := range networks {
+		ingressConnections = append(ingressConnections, net.IngressConnCount())
+		_, err := net.HealthCheck(context.Background())
+		healthCheckErrors = append(healthCheckErrors, err)
+	}
+
+	// First node has all nodes connected to it.
+	// Second node has only the third node connected to it.
+	// Third node has no node connected to it, as it connects to the first and second node.
+	require.Equal([]int{2, 1, 0}, ingressConnections)
+	require.Equal([]error{nil, nil}, healthCheckErrors[:2])
+	require.EqualError(healthCheckErrors[2], "network layer is unhealthy reason: primary network validator is unreachable")
```

Comment:
Can we add a TODO here to remove this string check? I'd love to not have this - but it seems like a bigger refactor than should be included in this PR.

Reply:
Hopefully I improved it by referencing a variable instead, [here](https://github.com/ava-labs/avalanchego/pull/3719/commits/9c9ca56e1ad115a5e5ba025a4df288e978e4ec12#diff-6534cc6f135999ac0a3c726bf471e812670d52b30707d8facd54fe5a61f9f58eL355).